com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -67,11 +63,24 @@ public StackingSensor(ISensor wrapped, int numStackedObservations)
             }
 
             // TODO support arbitrary stacking dimension
-            m_Shape[0] *= numStackedObservations;
+            m_Shape[m_Shape.Length - 1] *= numStackedObservations;
             m_StackedObservations = new float[numStackedObservations][];
-            for (var i = 0; i < numStackedObservations; i++)
+            m_StackedCompressedObservations = new byte[numStackedObservations][];
```

Comment:
nit: move this inside the `else` block (so that we don't allocate it if we don't need it). And same thing with m_StackedObservations in the `if` block